staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/promise/testing/counting.go

```diff
@@ -0,0 +1,107 @@
+/*
```

Comment:
I would suggest moving this test to fairqueuing/testing/promise.go

I think that keeping all test-related code in one place is much easier to follow than having is split across whole codebase.

Reply:
done, partly here and partly in #104212